File: lab2_task3/main.py
def count_sequences(sequence):
    if len(sequence) < 2:
        return 0
    
    seq_len = 1
    seq_count = 0
    
    for i in range(len(sequence) - 1):
        if sequence[i+1] >= sequence[i]:
            seq_len += 1
        elif seq_len > 1:
            seq_count += 1
            seq_len = 1
        else:
            seq_len = 1
    
    if seq_len > 1:
        seq_count += 1
    
    return seq_count

nums = list(map(int, input("Введите последовательность чисел через пробел: ").split()))
print(count_sequences(nums))

File: lab2_task3/main.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
	"strings"
)

func countSequences(sequence []int) int {
	if len(sequence) < 2 {
		return 0
	}

	seqLen := 1
	seqCount := 0

	for i := 0; i < len(sequence)-1; i++ {
		if sequence[i+1] >= sequence[i] {
			seqLen++
		} else if seqLen > 1 {
			seqCount++
			seqLen = 1
		} else {
			seqLen = 1
		}
	}

	if seqLen > 1 {
		seqCount++
	}

	return seqCount
}

func main() {
	reader := bufio.NewReader(os.Stdin)
	fmt.Println("Введите последовательность чисел через пробел:")
	input, _ := reader.ReadString('\n')
	input = strings.TrimSpace(input)
	parts := strings.Split(input, " ")
	
	var nums []int
	for _, s := range parts {
		num, _ := strconv.Atoi(s)
		nums = append(nums, num)
	}

	fmt.Println(countSequences(nums))
}